From 3505369c30d61916f6e25fb444fe78c04310aa46 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 8 Dec 2015 17:20:16 +0800
Subject: [PATCH 0784/5242] MLK-11920 video: fbdev: mxc ipuv3 fb: Use
 fb_set_var() to unblank a framebuffer

commit  f6a1a690794c80bc6dc71d77e1d6105a6f8d72f2 from
https://source.codeaurora.org/external/imx/linux-imx.git

The resolution of an overlay framebuffer could be at most the same to the
one of the relevant background framebuffer.  However, the resolution of a
HDMI monitor could be changed at runtime.  Thus, when we unblank the overlay
framebuffer, we should check it's resolution and start position in case it
exceeds the boundary of the background framebuffer.  This patch replaces
mxcfb_set_par() with fb_set_var() to implement the unblank operation so that
the logic contains the ->fb_check_var() path.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_ipuv3_fb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
index 38ea948..5535138 100644
--- a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
@@ -2338,7 +2338,7 @@ static int mxcfb_blank(int blank, struct fb_info *info)
 	case FB_BLANK_UNBLANK:
 		info->var.activate = (info->var.activate & ~FB_ACTIVATE_MASK) |
 				FB_ACTIVATE_NOW | FB_ACTIVATE_FORCE;
-		ret = mxcfb_set_par(info);
+		ret = fb_set_var(info, &info->var);
 		break;
 	}
 	if (!ret)
-- 
1.7.9.5

